File: 1. Two Sum/twoSum.cpp
#include <unordered_map>
#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> indices(2,0);
        unordered_map<int, int> hmap;
        int currIdx = 0;
        int key = 0;
        for(vector<int>::iterator it = nums.begin(); it != nums.end(); it++, currIdx++){
            key = target - it[0];
            if(hmap.find(it[0]) != hmap.end()){ //Found key
                indices[0] = hmap[it[0]];
                indices[1] = currIdx;
                cout<<"Here"<<currIdx<<";"<<endl;
                break;
            } else{
                hmap[key] = currIdx;
            }
        }
        return indices;
    }
};

File: 1. Two Sum/twoSum.py
from typing import *

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sumDict = {}
        res = []
        idx = 0
        for num in nums:
            key = target - num
            if(num in sumDict):
                res.append(sumDict[num])
                res.append(idx)
                break
            else:
                sumDict[key] = idx
            idx += 1
        return res

sol = Solution()
print(sol.twoSum([2,7,11,15], 9))